gnucash/gnome-utils/gnc-date-format.c

```diff
@@ -76,7 +76,7 @@ struct _GNCDateFormatPriv
 };
 
 #define GNC_DATE_FORMAT_GET_PRIVATE(o)  \
-   (G_TYPE_INSTANCE_GET_PRIVATE ((o), GNC_TYPE_DATE_FORMAT, GNCDateFormatPriv))
+   (G_TYPE_INSTANCE_GET_PRIVATE ((o), GNC_TYPE_DATE_FORMAT, GNCDateFormatPrivate))
```

Comment:
[Emmanuel Bassi's article](https://www.bassi.io/articles/2013/06/21/the-king-is-dead/) you refer to on the mailing list thread also suggests replacing calls to MY_OBJECT_GET_PRIVATE with my_object_get_instance_private calls.
While I wouldn't do it as part of this PR it may be a nice follow up cleanup action.